File: cpp/1st_Week/Methode/Titik.cpp
#include <iostream>
#include <string>

using namespace std;

class Titik
{
    /* Class yang digunakan untuk mengimplementasikan sebuah tipe titik */
    private:
        int x;      //koordinat x
        int y;      //koordinat y
    public:
    //Constructor
    Titik(){
        //di set 0 karena tidak ada parameter
        x = 0;
        y = 0;
    }

    //constructor dengan parameter
    Titik(int xp, int yp){
        x = xp;
        y = yp;
    }

    //fungsi set
   void setX(int xp){
        //mengset koordinat x
        x = xp;
    }

    void setY(int yp){
        //set koordinat y
        y = yp;
    }
    
    // fungsi get
    int getX(){
        return x;
    }

    int getY(){
        return y;
    }

    //destructor
    ~Titik(){
    }
};

File: java/javadpbo/src/com/sem4dpbo/week7/abstrakTitik/Titik.java
package com.sem4dpbo.week7.abstrakTitik;

abstract class Titik {
//    kelas untuk implementasi sebuah tipe titik
    private int x;
    private int y;

//    contructor
    Titik(){
        x = 0;
        y = 0;
    }

    Titik(int x, int y) {
        this.x = x;
        this.y = y;
    }

//    setter and getter X
    public int getX() {
        return x;
    }

    public void setX(int x) {
        this.x = x;
    }

//    setter and getter y
    public int getY() {
        return y;
    }

    public void setY(int y) {
        this.y = y;
    }

//    method abstract yang hanya di deklarasikan pada class anak
    public abstract void printTitik();
}
